AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -360,17 +360,23 @@ - (BOOL)removeAccount {
   return YES;
 }
 
-- (void)acquireTokenSilentlyWithMSALAccount:(MSALAccount *)account {
+- (void)acquireTokenSilentlyWithMSALAccount:(MSALAccount *)account withUIFallback:(BOOL)withUIFallback {
   __weak typeof(self) weakSelf = self;
   [self.clientApplication
       acquireTokenSilentForScopes:@[ (NSString * __nonnull) self.authConfig.authScope ]
                           account:account
-                  completionBlock:^(MSALResult *result, NSError *e) {
+                  completionBlock:^(MSALResult *result, NSError *error) {
                     typeof(self) strongSelf = weakSelf;
-                    if (e) {
-                      MSLogWarning([MSAuth logTag],
-                                   @"Silent acquisition of token failed with error: %@. Triggering interactive acquisition", e);
-                      [strongSelf acquireTokenInteractively];
+                    if (error) {
+                      if (withUIFallback && [error.domain isEqual:MSALErrorDomain] && error.code == MSALErrorInteractionRequired) {
+                        MSLogWarning([MSAuth logTag],
+                                     @"Silent acquisition of token failed with error: %@. Triggering interactive acquisition", error);
+                        [strongSelf acquireTokenInteractively];
+                      } else {
+                        [[MSAuthTokenContext sharedInstance] setAuthToken:nil withAccountId:nil expiresOn:nil];
+                        MSLogError([MSAuth logTag], @"User sign-in failed. Error: %@", error);
```

Comment:
Need more details in the log message in case MSAL returned `MSALErrorInteractionRequired` but `withUIFallback` is `NO`.